Sprint 11 - Commit # 1

DanteIT94 opened the pull request:
1) Was made first steps to understanding Multi-Threading

2) Was update user account by adding get-response into code

DanteIT94 committed:
Sprint 11 - Commit # 1

1) Was made first steps to understanding Multi-Threading

2) Was update user account by adding get-response into code

DanteIT94 committed:
Sprint 11 - Commit # 2

1) Was added profileservice updating in SplashViewController

2) minor fixes

DanteIT94 committed:
Sprint 11 - minor fixes commit

DanteIT94 committed:
Sprint 11 - KingFisher-Commit

DanteIT94 committed:
Update ProfileImageService.swift

DanteIT94 committed:
Sprint 11 _ pre-review commit

1) was deleted SplashScreen
2) was implimented Dependency -> KingFisher + KeyWrapper

DanteIT94 committed:
Sprint 11 - minor fixes

DanteIT94 committed:
Update SplashViewController.swift

DanteIT94 committed:
Sprint 11 - Folder's refactoring

andrew-kuzn-yndx commented on pinterestClone/AuthFolder + TokenRepository/ServiceAccessLayer(OAuth2)/OAuth2TokenStorage.swift:
```diff
@@ -0,0 +1,40 @@
+//        }
+//        set {
+//            return userDefaults.set(newValue, forKey: "token")
+//        }
```
Старайся не оставлять такое. Если не удалять мусор, то его может скопиться очень много, который будет мешать работать)

DanteIT94 replied:
Порой кажется что мусор может быть полезен) но по факту он всегда просто забивает обзор, а к старому коду ты обычно не возвращаешься :( 
Удалил, спасибо за помощь!

andrew-kuzn-yndx commented on pinterestClone/AuthFolder + TokenRepository/WebViewViewController.swift:
```diff
@@ -17,12 +17,13 @@ protocol WebViewViewControllerDelegate: AnyObject {
+    private var estimatedProgressObservation: NSKeyValueObservation?
     weak var delegate: WebViewViewControllerDelegate?
     @IBOutlet private var webView: WKWebView!
     @IBOutlet private var progressView: UIProgressView!
```
Молодец, что указал private для аутлетов! Также хорошей практикой является указание модификатора weak к IBOutlet. Вероятность retain cycle тут мала, но и нам ничего не стоит написать слабые ссылки)

DanteIT94 replied:
Спасибо за уточнение - сейчас перешел на верстку кодом)

andrew-kuzn-yndx commented on pinterestClone/AuthFolder + TokenRepository/WebViewViewController.swift:
```diff
@@ -42,36 +43,43 @@ final class WebViewViewController: UIViewController {
+//        } else {
+//            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
+//        }
+//    }
```
Половина контроллера состоит из мусора, это очень плохо. В процессе работы он может быть, но в рабочую ветку гита такое сливать нельзя

DanteIT94 replied:
Исправил, удалил

andrew-kuzn-yndx commented on pinterestClone/Helpers/URLSession - GenericFunc.swift:
```diff
@@ -0,0 +1,54 @@
+            ///проверяем наличие ошибок
+            guard error == nil else {
+                DispatchQueue.main.async {
+                    completion(.failure(NetworkError.urlRequestError(error!)))
```
Старайся избегать force unwrap, даже если уверен, что значение будет. Лучше использовать конструкции для безопасного разворачивания: guard let, if let

DanteIT94 replied:
Скорректировал!

andrew-kuzn-yndx commented on pinterestClone/Helpers/URLSession - GenericFunc.swift:
```diff
@@ -0,0 +1,54 @@
+                }
+            } catch {
+                DispatchQueue.main.async {
+                    completion(.failure(NetworkError.urlSessionError))
```
Молодец, что обработал ошибку. Но ведь тут ошибка декодинга, а не urlSessionError :)

DanteIT94 replied:
Сплоховал, не заметил. Спасибо за ремарку - исправил!

andrew-kuzn-yndx commented on pinterestClone/SplashView/SplashViewController.swift:
```diff
@@ -6,20 +6,25 @@
         } else {
-            /// Если токен не сохранен, значит пользователь не был ранее авторизован. Можно перенаправить на экран авторизации
-            performSegue(withIdentifier: ShowAuthSegueIdentifier, sender: nil)
+            //            performSegue(withIdentifier: ShowAuthSegueIdentifier, sender: nil)
```
Снова мусор

DanteIT94 replied:
"Выбросил"

andrew-kuzn-yndx commented on pinterestClone/SplashView/SplashViewController.swift:
```diff
@@ -30,40 +35,53 @@ class SplashViewController: UIViewController {
     private func switchToTabBarController() {
-        guard let window = UIApplication.shared.windows.first else { fatalError("Invalid Configuration") }
-        let tabBarController = UIStoryboard(name: "Main", bundle: .main)
+        guard let window = UIApplication.shared.windows.first else {
```
Со свойством window мы обычно работаем в SceneDelegate. Тут это не к месту

DanteIT94 replied:
Исправлю при рефакторинг кода!

andrew-kuzn-yndx commented on pinterestClone/SplashView/SplashViewController.swift:
```diff
@@ -73,18 +91,49 @@ extension SplashViewController: AuthViewControllerDelegate {
+                case .failure:
+                    let alert = UIAlertController(title: "Что-то пошло не так(", message: "Не удалось войти в систему", preferredStyle: .alert)
+                    alert.addAction(UIAlertAction(title: "Ок", style: .default, handler: nil))
+                    self.present(alert, animated: true, completion: nil)
```
Если тут будет ошибка, то UIBlockingProgressHUD не пропадет.
В этой функции вывзов UIBlockingProgressHUD нужен в любом случае, поэтому ни к чему помещать его к кейс, а лучше написать после свитч

DanteIT94 replied:
Пропустил этот момент( когда не тестируешь все варианты - сложно понятно( Спасибо за замечания!

andrew-kuzn-yndx commented on pinterestClone/TabBar/TabBarViewController.swift:
```diff
@@ -0,0 +1,29 @@
+            image: UIImage(named: "tab_profile_active"),
+            selectedImage: nil)
+//        storyboard.instantiateViewController(
+//            withIdentifier: "ProfileViewController")
```
Мусор

DanteIT94 replied:
"Выброшен"

andrew-kuzn-yndx commented on pinterestClone/UserProfileFolder/OldAPINotification(Useless).swift:
```diff
@@ -0,0 +1,65 @@
+//        let profileImageURL = userInfo["URL"] as? String,
+//        let url = URL(string: profileImageURL)
+//    else { return }
+//    
```
Неиспользуемые файлы и классы нужно полностью удалять из проекта

DanteIT94 replied:
"Выбросил"

andrew-kuzn-yndx commented on pinterestClone/UserProfileFolder/ProfileViewController.swift:
```diff
@@ -27,11 +40,11 @@ final class ProfileViewController: UIViewController {
-//        avatarImage.layer.borderWidth = 2
-//        avatarImage.layer.borderColor = UIColor.white.cgColor
+        //        avatarImage.layer.borderWidth = 2
+        //        avatarImage.layer.borderColor = UIColor.white.cgColor
```
опять :(

DanteIT94 replied:
"Выбросил"

DanteIT94 committed:
Sprint 11.5 - Transition to CodeLayout

1) was added AuthView and WebView with layout in code (without Storyboard)

DanteIT94 committed:
sorting trash from swift-files

DanteIT94 committed:
major fixes of OAuth2 + SingleImageView

1) was created SingleImageViewController by code
2) update OAuth2Service for best reading

DanteIT94 committed:
Code-Layout for whole App - Finished

DanteIT94 committed:
Merge pull request #5 from DanteIT94/sprint_11.5_storyboard_refactoring

Sprint 11.5 - Transition to CodeLayout

DanteIT94 committed:
Update AppDelegate.swift

Final update in sprint 11

DanteIT94 committed:
Update SplashViewController.swift

andrew-kuzn-yndx commented on pinterestClone/AuthFolder + TokenRepo(SKW)/AuthViewController.swift:
```diff
@@ -0,0 +1,89 @@
+            enterButton.heightAnchor.constraint(equalToConstant: 48),
+            enterButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            enterButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            enterButton.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -90)//возможны корректировки
```
Молодец, что этот контроллер тоже решил делать кодом :)

andrew-kuzn-yndx commented on pinterestClone/AuthFolder + TokenRepo(SKW)/WebViewViewController.swift:
```diff
@@ -0,0 +1,165 @@
+    }
+    
+    ///функция code(from:) - она возвращает код авторизации, если он получен
+    private func code(from navigationAction: WKNavigationAction) -> String? {
```
В целом ничего страшного, но будто это должно быть не в контроллере, а в сервисном слое..
Если будет желание, попробуй перенести это в другую зону ответственности)

DanteIT94 replied:
Постараюсь в рамках 12 спринта это сделать!

andrew-kuzn-yndx commented on pinterestClone/Helpers/URLSession - GenericFunc.swift:
```diff
@@ -7,48 +7,44 @@
+                        let decodedModel = try jsonDecoder.decode(T.self, from: data)
+                        fulfilCompletion(.success(decodedModel))
+                    } catch {
+                        fulfilCompletion(.failure(NetworkError.decodingError))
```
Молодец!

andrew-kuzn-yndx commented on pinterestClone/ImagesListFolder/ImagesListCell.swift:
```diff
@@ -9,32 +9,90 @@ import UIKit
-        //TODO: - градиентный слой (✅DONE)
+    
+    //MARK: - Private Computered Properties
+    private let dateLabel: UILabel = {
```
Решил забить на weak с аутлетами и решил вообще всё сделать кодом 😀 красавчик!

DanteIT94 replied:
Кодом рисовать оказалось интереснее)

andrew-kuzn-yndx commented on pinterestClone/ImagesListFolder/ImagesListCell.swift:
```diff
@@ -9,32 +9,90 @@ import UIKit
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        assertionFailure("Error")
```
Молодец, что не использовал fatalError!

DanteIT94 replied:
Чтобы избежать ненужных крашей!

DanteIT94 committed:
Try to add Self-Made KeyChainManager (will be added during 12 Sprint)
